src/blobstreamx.cairo

```diff
@@ -0,0 +1,362 @@
+                        trusted_block: latest_block,
+                        trusted_header: latest_header,
+                        target_block: _target_block
+                    }
+                );
+        }
+
+        /// Commits the new header at targetBlock and the data commitment for the block range [trustedBlock, targetBlock).
+        ///
+        /// # Arguments 
+        ///
+        /// * `_target_block` -  The end block of the header range request
+        fn commit_header_range(ref self: ContractState, _target_block: u64) {
+            let latest_block = self.get_latest_block();
+            let trusted_header = self.block_height_to_header_hash.read(latest_block);
+            assert(trusted_header != 0, TendermintXErrors::TrustedHeaderNotFound);
+
+            assert(_target_block > latest_block, TendermintXErrors::TargetBlockNotInRange);
+            assert(
+                _target_block - latest_block <= self.data_commitment_max(),
+                TendermintXErrors::TargetBlockNotInRange
+            );
+
+            let mut input = BytesTrait::new_empty();
+            input.append_u64(latest_block);
+            input.append_u256(trusted_header);
+            input.append_u64(_target_block);
+
+            let (target_header, data_commitment) = ISuccinctGatewayDispatcher {
+                contract_address: self.get_gateway()
+            }
+                .verified_call(self.get_header_range_id(), input);
+
+            let proof_nonce = self.get_state_proof_nonce();
+            self.block_height_to_header_hash.write(_target_block, target_header);
+            self.state_data_commitments.write(proof_nonce, data_commitment);
+            self
+                .emit(
+                    DataCommitmentStored {
+                        proof_nonce,
+                        data_commitment,
+                        start_block: latest_block,
+                        end_block: _target_block,
+                    }
+                );
+            self.emit(HeadUpdate { target_block: latest_block, target_header: target_header });
+            self.state_proof_nonce.write(proof_nonce + 1);
+            self.latest_block.write(_target_block);
+        }
+
+
+        /// Prove the validity of the next header and a data commitment for the block range [latestBlock, latestBlock + 1).
+        fn request_next_header(ref self: ContractState) {
+            let latest_block = self.get_latest_block();
+            let latest_header = self.block_height_to_header_hash.read(latest_block);
+            assert(latest_header != 0, TendermintXErrors::LatestHeaderNotFound);
+
+            let mut input = BytesTrait::new_empty();
+            input.append_u64(latest_block);
+            input.append_u256(latest_header);
+
+            let mut entry_calldata = BytesTrait::new_empty();
+            entry_calldata.append_felt252(selector!("commit_next_header"));
+            input.append_u64(latest_block);
```

Comment:
Should this be appending to `entry_calldata`?

Reply:
good catch